From 5d9b73279f0c9c020823b0854a72eb5f474d1234 Mon Sep 17 00:00:00 2001
From: Philby John <pjohn@mvista.com>
Date: Tue, 7 Sep 2010 17:11:58 +0530
Subject: [PATCH] Fix: "Fatal exception: std::exception" errors

Source: MontaVista Software, LLC
MR: 24839
Type: Defect Fix
Disposition: Needs submitting to http://sourceforge.net/projects/tripwire/
ChangeID: 7c9a0f430a700376ac7cf7752400eb3df6c98097
Description:
This corrects random tripwire failure with the error message
"Fatal exception: std::exception".Possible cause may be last byte missed
in compressed report file.

Signed-off-by: Philby John <pjohn@mvista.com>
---
 src/cryptlib/queue.cpp |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/cryptlib/queue.cpp b/src/cryptlib/queue.cpp
index 6501128..ab5d787 100644
--- a/src/cryptlib/queue.cpp
+++ b/src/cryptlib/queue.cpp
@@ -67,7 +67,11 @@ unsigned int ByteQueueNode::Put(const byte *inString, unsigned int length)
 
 unsigned int ByteQueueNode::Get(byte &outByte)
 {
-	if (tail==head)
+	if(tail == 0)
+		return 1;
+	else if ((tail < 256) && (head == tail))
+		return 1;
+	else if (tail==head)
 		return 0;
 
 	outByte=buf[head++];
-- 
1.6.3.3.333.g4d53f

